src/main/io/gps.c

```diff
@@ -971,7 +985,11 @@ void gpsConfigureNmea(void)
                     break;
                 case 1: // second run
                     // print the init string for the baudrate we want to be at
+#ifdef USE_GPS_SIRF
+                    serialPrint(gpsPort, "$PSRF100,0,115200,8,1,0*04\r\n");
+#else
                     serialPrint(gpsPort, "$PSRF100,1,115200,8,1,0*05\r\n");
+#endif
```

Comment:
It doesn't seem correct to be using the proprietary SiRF Tech command by default?
Will u-blox (or any other manufacturer's) modules respond to Sirf commands (I don't think so)?

With the changes suggested around line 128 above, wouldn't we just want to send the string in `gpsInitData[n].nmea` (currently `mtk`)?

In general, I think the procedure for establishing the baud rate and connection for NMEA should be similar to what we worked out for UBX. As I understand it, the module itself isn't in NMEA or UBX "mode", it will respond the same, it's just a matter of which protocols we enable for the ports and which commands we use.

Reply:
If that is the case we can remove the mtk (nmea) part and only use `PUBX,41` message. That's why I added some custom defines for testing different settings.

Reply:
Perhaps, I suspect someone had a MediaTek GPS at some point, or one that implemented the MediaTek proprietary command set, and that's why it was implemented using those?

I was incorrect about GlobalTop Tech, they are not the MTK command originators, that is MediaTek (hence MTK).
Searching around more, it seems that multiple manufacturers have adopted some of the proprietary command sets, like MTK and SRF.

I really don't know what the BF user base looks like in terms of those who want to use NMEA protocol and what hardware they have? I would think most BF users have u-blox based modules, but not sure that's true of BF NMEA users?

I can do some more homework and try to figure out what the most popular proprietary NMEA command sets are, but it's such a broad overall user base that I doubt it maps to the BF user base.

Reply:
Refactored taking into account your review points.